tests/eko/io/test_init.py

```diff
@@ -0,0 +1,285 @@
+    for key, value in a.items():
+        if isinstance(value, dict):
+            assert sorted(value.keys()) == sorted(b[key].keys())
+
+
+class TestLegacy:
+    def test_io(self, eko_factory: EKOFactory, tmp_path):
+        # create object
+        o1 = eko_factory.get()
+        # test streams
+        stream = io.StringIO()
+        legacy.dump_yaml(o1, stream)
+        # rewind and read again
+        stream.seek(0)
+        o2 = legacy.load_yaml(stream)
+        xgrid = eko_factory.operator.rotations.xgrid
+        np.testing.assert_almost_equal(o1.xgrid.raw, xgrid.raw)
+        np.testing.assert_almost_equal(o2.xgrid.raw, xgrid.raw)
+        # fake eko.io files
+        fpyaml = tmp_path / "test.yaml"
+        legacy.dump_yaml_to_file(o1, fpyaml)
+        # fake input file
+        o3 = legacy.load_yaml_from_file(fpyaml)
+        np.testing.assert_almost_equal(o3.xgrid.raw, xgrid.raw)
+        # repeat for tar
+        fptar = tmp_path / "test.tar"
+        legacy.dump_tar(o1, fptar)
+        o4 = legacy.load_tar(fptar)
+        np.testing.assert_almost_equal(o4.xgrid.raw, xgrid.raw)
+        fn = "test"
+        with pytest.raises(ValueError, match="wrong suffix"):
+            legacy.dump_tar(o1, fn)
+
+    def test_rename_issue81(self):
+        # https://github.com/N3PDF/eko/issues/81
+        # create object
+        o1, fake_card = fake_legacy
+        for q2, op in fake_card["Q2grid"].items():
+            o1[q2] = eko.io.Operator.from_dict(op)
+
+        with tempfile.TemporaryDirectory() as folder:
+            # dump
+            p = pathlib.Path(folder)
+            fp1 = p / "test1.tar"
+            fp2 = p / "test2.tar"
+            legacy.dump_tar(o1, fp1)
+            # rename
+            shutil.move(fp1, fp2)
+            # reload
+            o4 = legacy.load_tar(fp2)
+            np.testing.assert_almost_equal(
+                o4.xgrid.raw, fake_card["interpolation_xgrid"]
+            )
+
+    def test_io_bin(self):
+        # create object
+        o1, fake_card = fake_legacy
+        for q2, op in fake_card["Q2grid"].items():
+            o1[q2] = eko.io.Operator.from_dict(op)
+        # test streams
+        stream = io.StringIO()
+        legacy.dump_yaml(o1, stream, False)
+        # rewind and read again
+        stream.seek(0)
+        o2 = legacy.load_yaml(stream)
+        np.testing.assert_almost_equal(o1.xgrid.raw, fake_card["interpolation_xgrid"])
+        np.testing.assert_almost_equal(o2.xgrid.raw, fake_card["interpolation_xgrid"])
+
+
+class TestManipulate:
+    def test_xgrid_reshape(self, eko_factory: EKOFactory):
+        # create object
+        xg = interpolation.XGrid(np.geomspace(1e-5, 1.0, 21))
+        o1 = eko_factory.get()
+        o1.xgrid = xg
+        o1.rotations._targetgrid = xg
+        o1.rotations._inputgrid = xg
+        o1._operators = {
+            10: eko.io.Operator.from_dict(
+                dict(
+                    operator=eko_identity([1, 2, len(xg), 2, len(xg)])[0],
```

Comment:
It is, and this file contains the only tests still failing (if you look around, there are still a couple of occurrences of `fake_legacy`).

In particular, the current way to do this properly is to set the actual grids you want to use in the `eko_factory` runcards, before `get`ting it. The operators you still need to fix afterwards, since they are randomly generated.